generated/nitclk/session.py

```diff
@@ -0,0 +1,1156 @@
+import ctypes
+import datetime
+import threading
+
+import nitclk._attributes as _attributes
+import nitclk._converters as _converters
+import nitclk._library_singleton as _library_singleton
+import nitclk._visatype as _visatype
+import nitclk.errors as errors
+
+# Used for __repr__ and __str__
+import pprint
+pp = pprint.PrettyPrinter(indent=4)
+
+_session_instance = None
+_session_instance_lock = threading.Lock()
+
+
+# Helper functions for creating ctypes needed for calling into the driver DLL
+def get_ctypes_pointer_for_buffer(value=None, library_type=None, size=None):
+    if isinstance(value, array.array):
+        assert library_type is not None, 'library_type is required for array.array'
+        addr, _ = value.buffer_info()
+        return ctypes.cast(addr, ctypes.POINTER(library_type))
+    elif str(type(value)).find("'numpy.ndarray'") != -1:
+        import numpy
+        return numpy.ctypeslib.as_ctypes(value)
+    elif isinstance(value, list):
+        assert library_type is not None, 'library_type is required for list'
+        return (library_type * len(value))(*value)
+    else:
+        if library_type is not None and size is not None:
+            return (library_type * size)()
+        else:
+            return None
+
+
+class SessionReference(object):
+    '''Properties container for NI-TClk attributes.'''
+
+    # This is needed during __init__. Without it, __setattr__ raises an exception
+    _is_frozen = False
+
+    exported_sync_pulse_output_terminal = _attributes.AttributeViString(2)
+    '''Type: str
+
+    Specifies the destination of the Sync Pulse. This property is most often  used when synchronizing a multichassis system.
+    Values
+    Empty string. Empty string is a valid value, indicating that the signal is  not exported.
+    PXI Devices -  'PXI_Trig0' through  'PXI_Trig7' and device-specific settings
+    PCI Devices -  'RTSI_0' through  'RTSI_7' and device-specific settings
+    Examples of Device-Specific Settings
+    - NI PXI-5122 supports  'PFI0' and  'PFI1'
+    - NI PXI-5421 supports  'PFI0',  'PFI1',  'PFI4', and  'PFI5'
+    - NI PXI-6551/6552 supports  'PFI0',  'PFI1',  'PFI2', and  'PFI3'
+    Default Value is empty string
+    '''
+    exported_tclk_output_terminal = _attributes.AttributeViString(9)
+    '''Type: str
+
+    Specifies the destination of the device's TClk signal.
+    Values
+    Empty string. Empty string is a valid value, indicating that the signal is  not exported.
+    PXI Devices -  'PXI_Trig0' through  'PXI_Trig7' and device-specific settings
+    PCI Devices -  'RTSI_0' through  'RTSI_7' and device-specific settings
+    Examples of Device-Specific Settings
+    - NI PXI-5122 supports  'PFI0' and  'PFI1'
+    - NI PXI-5421 supports  'PFI0',  'PFI1',  'PFI4', and  'PFI5'
+    - NI PXI-6551/6552 supports  'PFI0',  'PFI1',  'PFI2', and  'PFI3'
+    Default Value is empty string
+    '''
+    pause_trigger_master_session = _attributes.AttributeViInt32SessionReference(6)
```

Comment:
I think this class should be `AttributeSessionReference`, since that's the Python-facing type.

Reply:
Changed